### PR TITLE
Fix one protential memory leak

### PR DESCRIPTION
**Pull request by js0701**
Delete a created surface

Change-Id: I2d0dc2195b6e2646c632dcd84e060c72cb97c108
Tracked-On: None
Tests: Android UI normal
Signed-off-by: Lin Johnson <johnson.lin@intel.com>

### CODE REVIEW

**Comment by js0701** [comment]
@yhe39 
Can you review and validate
Thanks

**Comment by yhe39** [comment]
hi @js0701 the patch https://github.com/intel/IA-Hardware-Composer/pull/546/commits/cf381523eaafad87f19842b3e3ffaf7c2647654b
will cause one regression: in acrn when record screen the HWC will reboot causing the UI reboot.

**Comment by js0701** [comment]
@yhe39 
Thanks.
I need to figure out why:)

**Comment by js0701** [comment]
@yhe39 
I can't figure out why the first patch is causing reboot when virtual display is used

Do you know?
If you don't,
Maybe can you set up some debug about it.
From the source code, delete surface is necessary

**Comment by yhe39** [comment]
ok, i will check on this. Thanks. 
On baremetal, there is no such issue. So the harman do not have the reboot issue.
Only on acrn have the reboot issue.

**Comment by yhe39** [comment]
With the new delete handle_ patch, do not have the ui reboot issue.

**Comment by js0701** [comment]
@yhe39  Can you run some test and if no regression please go with merge

File: common/display/virtualdisplay.h [hunk 1]
```diff
@@ -95,13 +95,12 @@ class VirtualDisplay : public NativeDisplay {
...
 
  private:
-  HWCNativeHandle output_handle_;
+  HWCNativeHandle output_handle_ = 0;
   int32_t acquire_fence_ = -1;
   Compositor compositor_;
   uint32_t width_ = 1;
   uint32_t height_ = 1;
   std::vector<OverlayLayer> in_flight_layers_;
-  HWCNativeHandle handle_ = 0;
```
Comment (YuanjunHuang) [inline common/display/virtualdisplay.h]: seems the handle_ was useless.

**Comment by js0701** [comment]
@yhe39  Please approve this PR if no more concerns
@Shao-Feng  Will merge

**Comment by yhe39** [comment]
hi @js0701 i apply the 3 patches on latest github code and test, **still has the ui reboot issue**. it is the code "delete surface" cause the reboot. i retest many times, have this issue indeed. seems the delete 
handle_ patch doesn't resolve this issue. Do NOT know why yesterday it works. Sorry for the mistake:(.

**Comment by js0701** [comment]
@yhe39  Good. I also think the 3rd patch can't fix the problem.
We need to work more

**Comment by yhe39** [comment]
hi, in the Compositor::DrawOffscreen,  draw_state.destroy_surface_ = true;
In CompositorThread::Handle3DDrawRequest(), 
    if (draw_state.destroy_surface_) {
      if (draw_succeeded_) {
        draw_state.retire_fence_ =
            draw_state.surface_->GetLayer()->ReleaseAcquireFence();
      }
      delete draw_state.surface_;
    }
So i think the surface created in compositor is deleted properly.
And i try to change to only create one surface and reuse the surface for VDS, the screen record only has the first frame, and has the error: 
11-11 11:15:00.102  7029 10403 E iahwcomposer: Handle3DDrawRequest: Failed to Draw: error: Success
11-11 11:15:00.102  7029  7029 E iahwcomposer: Present: Failed to prepare for the frame composition ret=0
11-11 11:15:00.102  7029  7029 E iahwcomposer: Failed to set layers in the composition
Thanks!

**Comment by js0701** [comment]
@Shao-Feng  Please merge the 2nd and 3rd patch first. We are still in debugging with the first one

**Comment by Shao-Feng** [comment]
OK, I will cherry-pick them into another PR

**Comment by js0701** [comment]
@yhe39  Thanks for the comments. Let's have this closed

**Pull request closed without merging**